core/src/test/java/org/elasticsearch/discovery/DiscoveryWithServiceDisruptionsIT.java

```diff
@@ -1044,12 +1049,17 @@ public void testReadOnPostRecoveryShards() throws Exception {
             Future<ClusterRerouteResponse> rerouteFuture = internalCluster().client().admin().cluster().prepareReroute().add(new MoveAllocationCommand(new ShardId("test", 0), node_1, node_2)).setTimeout(new TimeValue(1000, TimeUnit.MILLISECONDS)).execute();
 
             logger.info("--> wait for relocation to start");
-            // wait for relocation to start
-            beginRelocationLatchNode2.await();
+
             // start to block cluster state updates on node_1 and node_2 so that we end up with two primaries
             // one STARTED on node_1 and one in POST_RECOVERY on node_2
-            disruptionNode1.startDisrupting();
+            // wait for relocation to start on node_2
+            beginRelocationLatchNode2.await();
```

Comment:
That was indeed unintentional. I changed now to block once the files info request is received.